runtime/dotnet/core/ComposerLuisExtensions.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.BotFramework.Composer.Core
+{
+    public static class ComposerLuisExtensions
+    {
+        /// <summary>
+        /// Setup configuration to utilize the settings file generated by bf luis:build. This is a luis settings extensions adapter aligning with Composer customized settings.
+        /// </summary>
+        /// <remarks>
+        /// This will pick up LUIS_AUTHORING_REGION or --region settings as the setting to target.
+        /// This will pick up --environment as the environment to target.  If environment is Development it will use the name of the logged in user.
+        /// This will pick up --root as the root folder to run in.  
+        /// </remarks>
+        /// <param name="builder">Configuration builder to modify.</param>
+        /// <returns>Modified configuration builder.</returns>
+        public static IConfigurationBuilder UseComposerLuisSettings(this IConfigurationBuilder builder)
+        {
+            var configuration = builder.Build();
+            var botRoot = configuration.GetValue<string>("bot") ?? ".";
+            var luisRegion = configuration.GetValue<string>("LUIS_AUTHORING_REGION") ?? configuration.GetValue<string>("luis:authoringRegion") ?? configuration.GetValue<string>("luis:region") ?? "westus";
+            var environment = configuration.GetValue<string>("luis:environment") ?? Environment.UserName;
+            var settings = new Dictionary<string, string>();
+            settings["luis:endpoint"] = configuration.GetValue<string>("luis:endpoint") ?? $"https://{luisRegion}.api.cognitive.microsoft.com";
+            settings["BotRoot"] = botRoot;
+            builder.AddInMemoryCollection(settings);
+            if (environment == "Development")
+            {
+                environment = Environment.UserName;
+            }
+
+            var di = new DirectoryInfo(botRoot);
+            foreach (var file in di.GetFiles($"luis.settings.{environment.ToLower()}.{luisRegion}.json", SearchOption.AllDirectories))
```

Comment:
Do we need 'allDirectories'? File I/O is pretty slow and it would affect load time for scenarios like functions where load happens more frequently. We should minimize I/O as much as possible here.

Reply:
For example, iterating through our bin and obj folders is just a no-go. Can we be more specific?

Reply:
Also, the `ResourceExplorer` instance in the startup already has very optmized code to find files. Can you get the luis settings using the resource explorer? What changed that we need this search? Thanks in advance, I'd like to understand more in general about this scenario.